gmso/formats/lammpsdata.py

```diff
@@ -199,3 +209,228 @@ def write_lammpsdata(topology, filename, atom_style='full'):
+
+
+def read_lammpsdata(filename, atom_style='full', unit_style='real', potential='lj'):
+    """
+    Read in a lammps data file as a GMSO topology
+
+    Parameters
+    ----------
+    filename : str
+        LAMMPS data file
+    atom_style : str, optional, default='full'
+        Inferred atom style defined by LAMMPS
+    potential: str, optional, default='lj'
+        Potential type defined in data file
+
+    Returns
+    -------
+    top : GMSO Topology
+        A GMSO Topology object
+
+    Notes
+    -----
+    See http://lammps.sandia.gov/doc/2001/data_format.html for a full description of the LAMMPS data format.  
+    Currently only supporting LJ potential parameters.
+    """
+    # TODO: Add argument to ask if user wants to infer bond type
+    top = Topology()
+
+    # Validate 'atom_style'
+    if atom_style not in ['full']:
+        raise ValueError('Atom Style "{}" is invalid or is not currently supported'.format(
+                atom_style))
+
+    # Validate 'unit_style'
+    if unit_style not in ['real']:
+        raiseValueError('Unit Style "{}" is invalid or is not currently supported'.format(
+            unit_style))
+
+    # Parse box information
+    _get_box_coordinates(filename, unit_style, top)
+    # Parse atom type information
+    top, type_list = _get_ff_information(filename, unit_style, top)
+    # Parse atom information
+    _get_atoms(filename, top, unit_style, type_list)
+    # Parse connection (bonds, angles, dihedrals) information
+    # TODO: Add more atom styles
+    if atom_style in ['full']:
+        _get_connection(filename, top, unit_style, connection_type='bond')
+        _get_connection(filename, top, unit_style, connection_type='angle')
+
+    return top
+
+def get_units(unit_style):
+    """
+    Get units for specific LAMMPS unit style
+    """
+    unit_style_dict = {
+            'real': {'mass': u.g,
+                      'distance': u.angstrom,
+                      'energy': u.kcal/u.mol,
+                      'angle': u.radian,
+                      'charge': u.elementary_charge
+                      }
+            }
+
+    return unit_style_dict[unit_style]
+
+
+def _get_connection(filename, topology, unit_style, connection_type):
+    """
+    General function to parse connection types
+    """
+    with open(filename, 'r') as lammps_file:
+        for i, line in enumerate(lammps_file):
+            if connection_type in line.split():
+                n_connection_types = int(line.split()[0])
+            else:
+                return topology
+            if connection_type.capitalize() in line.split():
+                break
+    connection_type_lines = open(filename, 'r').readlines()[i+2:i+n_connection_types+2]
+    connection_type_list = list()
+    for line in connection_type_lines:
+        if connection_type == 'bond':
+            c_type = BondType(name=line.split()[0]
+                    )
+            c_type.parameters['k']=float(line.split()[1])*u.Unit(
```

Comment:
maybe a small comment that you are converting k/2 to k?

was not immediately obvious when i looked at it